int64 wheel
int64 velocity
